nixos/modules/misc/documentation.nix

```diff
@@ -36,6 +60,43 @@ let
           )
           pkgSet;
       in scrubbedEval.options;
+    baseOptionsJSON =
+      let
+        filter =
+          builtins.filterSource
+            (n: t:
+              (t == "directory" -> baseNameOf n != "tests")
+              && (t == "file" -> hasSuffix ".nix" n)
+            );
+      in
+        pkgs.runCommand "lazy-options.json" {
+          libPath = filter "${toString pkgs.path}/lib";
+          pkgsLibPath = filter "${toString pkgs.path}/pkgs/pkgs-lib";
+          nixosPath = filter "${toString pkgs.path}/nixos";
```

Comment:
copying the full `pkgs.path` is really expensive, to the point where it's not worth the effort of splitting the docs build at all. if there was a "use this store path exactly without copying" we'd love to use that instead of copying peacemeal, but that wouldn't help use cases that pin nixpkgs to a non-store path either :(

Reply:
My understanding of this code was wrong at first.
I think we can optimize the case of an already in-store `pkgs.path`, such as with flakes, niv or really any nix-based pinning method. In this case, we don't have to copy anything, as the path is already in the store, ready to be used in a build. e.g.

```nix
if lib.isStorePath pkgs.path
  then "${builtins.storePath pkgs.path}/lib"
  else filter "${toString pkgs.path}/lib"
```

`builtins.storePath` is supported under both pure and restricted evaluation. It basically adds the store path to the context rather than adding a _new_ directory to the store.

Reply:
nice! that's exactly what we were looking for and missed. the docs do however say that it's *not* available in pure evaluations, but it seems to work fine for flakes builds regardless. thanks for the pointer! :+1: